src/ui/components/RecipeEnrollmentPage.tsx

```diff
@@ -0,0 +1,72 @@
+import { ChangeEvent, FC, useState, useCallback, useEffect } from "react";
+import { Form, Container, Button } from "react-bootstrap";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+const MainPage: FC = () => {
+  const [jsonInput, setJsonInput] = useState("");
+  const [rows, setRows] = useState(19);
+
+  const handleInputChange = useCallback(
+    (event: ChangeEvent<HTMLTextAreaElement>) => {
+      setJsonInput(event.target.value);
+    },
+    [],
+  );
+
+  const handleEnrollClick = useCallback(async () => {
+    try {
+      const result = await browser.experiments.nimbus.enrollInExperiment(
+        JSON.parse(jsonInput) as object,
+      );
+
+      if (result) {
+        console.log("Enrollment successful");
+        alert("Enrollment successful");
+        setJsonInput("");
+      } else {
+        console.log("Enrollment failed");
+        alert("Enrollment failed");
+      }
+    } catch (error) {
+      console.error(error);
+      alert(error);
+    }
+  }, [jsonInput, setJsonInput]);
+
+  const updateRows = () => {
+    setRows(Math.max(2, (window.innerHeight - 75) / 30));
+  };
+
+  useEffect(() => {
+    window.addEventListener("resize", updateRows);
+    updateRows();
+    return () => {
+      window.removeEventListener("resize", updateRows);
+    };
+  }, []);
```

Comment:
Even if we do want to do this, we don't want to include the same logic and set of constants multiple times so we'd want to pull this out into a little utility file.